tools/triage/clean_old_tickets_attachments.py

```diff
@@ -0,0 +1,172 @@
+) -> list[jira.Issue]:
+    """Fetches and returns all issues from a given Jira project which created at least created_before days.
+
+    Args:
+        jira_client: Jira client for interacting with Jira API.\n
+        jira_project_key: Jira Project's key to fetch issues from.\n
+        logger: logger.\n
+        created_before: Created before at least created_before days old.\n
+        issues_limit: Maximum number of issues.\n
+
+    Returns:
+        list of matching issues.\n
+    """
+
+    result_list = jira_client.search_issues(
+        f"project = {jira_project_key} AND attachments IS NOT EMPTY AND created <= -{created_before}d",
+        maxResults=issues_limit,
+    )
+    logger.info(f"Found {len(result_list)} issues")
+
+    return list(result_list)
+
+
+def update_issues(jira_client: jira.JIRA, issues: list[jira.Issue], logger: logging.Logger):
+    """Removes watchers & attachments from given issues and then closes them.
+
+    Args:
+        jira_client: Jira client for interacting with Jira API.\n
+        issues: Given issues.\n
+        logger: logger.\n
+
+    Raises:
+        ExceptionGroup: when jira client fails to remove a watcher or attachment from an issue, or when it fails to close an issue.
+    """
+
+    for issue in issues:
+        errors = []
+
+        for watcher in jira_client.watchers(issue).watchers:
+            try:
+                jira_client.remove_watcher(issue.key, watcher.name)
+                logger.debug(f"Removed watcher {watcher.name} from issue {issue.key}")
+            except jira.JIRAError as e:
+                logger.debug(f"Failed to remove watcher {watcher.name} from issue {issue.key}: {str(e)}")
+                errors.append(e)
+
+        if errors:
+            for error in errors:
+                logger.error(f"error while removing watchers from {issue.key}", exc_info=error)
+
+            raise ExceptionGroup(f"failed removing all watchers from {issue.key}", errors)  # noqa: F821
+
+        errors = []
+
+        for attachment in issue.fields.attachment:
+            try:
+                attachment.delete()
+                logger.debug(f"Removed attachment {attachment.filename} from issue {issue.key}")
+            except jira.JIRAError as e:
+                logger.debug(f"Failed to remove attachment {attachment.filename} from issue {issue.key}: {str(e)}")
+                errors.append(e)
+
+        if errors:
+            for error in errors:
+                logger.error(f"error while removing attachments from {issue.key}", exc_info=error)
+
+            raise ExceptionGroup(f"failed removing all attachments from {issue.key}", errors)  # noqa: F821
+
+        errors = []
+
+        try:
+            jira_client.transition_issue(issue.key, CLOSED_STATUS)
+            logger.debug(f"Closed issue {issue.key}")
+        except jira.JIRAError as e:
+            logger.error(f"Failed to close issue {issue.key}: {str(e)}", exc_info=e)
```

Comment:
let's be consistent about the logs of the operations (currently it's sometimes error sometimes debug). I think you can do:
```suggestion
            logger.exception(f"Failed to close issue {issue.key}: {str(e)}")
```

Reply:
Sorry about that, it is a mistake. IMHO, I think we should `logger.debug` all exceptions during the executions and at the end `logger.exception` all of them similar to - https://github.com/openshift-assisted/assisted-installer-deployment/blob/d0948f1e2089a7ac98378c725bb2ed5245fa33cb/tools/triage/attach_logs_in_jira.py#L93-L97.
WDYT ?